### DISCUSSION TITLE
Remotely clone window

### DISCUSSION THREAD

**Discussion by infogrind** [kovidgoyal/kitty#6860] [General]
I would like to remote control Kitty to perform the following events (so that I can trigger it from the MacOS Alfred App):

1. Create a new OS window
2. Connect to a remote machine
3. cd to a specific directory
4. Create a new window that is a "clone" of the existing one, i.e., already connected to the remote machine and in the right directory.

Step 4 is the same that happens when I manually press the keyboard shortcut for `new_window_with_cwd` when being connected via SSH kitten.

I can perform steps 1 and 2 by running
```
kitty @ launch --type os-window --title mywindow kitty + kitten ssh <hostname>
```

But when I do
```
kitty @ launch --match 'title:mywindow'
```

this creates a new window, but the window is not inheriting the SSL session.

My overall goal is to define an Action in Alfred that opens a new Kitty OS window with two windows, both connected via SSH to my remote developer workstation and both in the same directory.

I would be glad for any hints how I can achieve this.

**Comment by kovidgoyal** [comment]
Run

kitty @ launch --type os-window --title mywindow kitty + kitten ssh <hostname>

twice

On Mon, Nov 27, 2023 at 02:41:30AM -0800, Marius Kleiner wrote:
> I would like to remote control Kitty to perform the following events (so that I can trigger it from the MacOS Alfred App):
> 
> 1. Create a new OS window
> 2. Connect to a remote machine
> 3. cd to a specific directory
> 4. Create a new window that is a "clone" of the existing one, i.e., already connected to the remote machine and in the right directory.
> 
> Step 4 is the same that happens when I manually press the keyboard shortcut for `new_window_with_cwd` when being connected via SSH kitten.
> 
> I can perform steps 1 and 2 by running
> ```
> kitty @ launch --type os-window --title mywindow kitty + kitten ssh <hostname>
> ```
> 
> But when I do
> ```
> kitty @ launch --match 'title:mywindow'
> ```
> 
> this creates a new window, but the window is not connected.
> 
> I would be glad for any hints how I can achieve this.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6860
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by infogrind** [reply]
Thanks, that is super simple.

If I also want to change the directory in both of them, I can do it using `kitten @ send-text ... cd <mydir>`.  But that makes the `cd <mydir>` command appear in the window,  as if I had typed it. Is there a way how this can be done silently, such that the change of directory on the SSH server happens "behind the scenes"?

An alternative way of asking the question is maybe: Could a sequence of `kitten @` be put in a script file that could be externally triggered, as in
```
kitty @ --to unix:/tmp/kitty.sock -f <myscript>
```
where the file `<myscript>` contains lines like
```
launch …
send-text …
```

Also, I wanted to say big thank you – with Kitty and with all the support you provide for it you are making a huge gift to the community. I am completely aware that any question you answer is an additional gift. Thanks 🙏

**Comment by kovidgoyal** [comment]
The ssh kitten allows you to control many things including the start
dir, read its docs.

**Reply by infogrind** [reply]
Thanks, will do!
